compliant/src/lib.rs

```diff
@@ -0,0 +1,1171 @@
+                }
+            }
+        } else {
+            // Compact output
+            #[cfg(feature = "enable_rocm")]
+            {
+                println!("│   ├── {} CUDA", cuda_mark);
+                println!("│   ╰── {} ROCM", rocm_mark);
+            }
+
+            #[cfg(not(feature = "enable_rocm"))]
+            {
+                println!("│   ╰── {} CUDA", cuda_mark);
+            }
+        }
+
+        // ABI status section
+        println!("╰── abi: {}", abi_status);
+        println!("    ├── {} {}", abi_mark, abi_output.manylinux_version);
+        println!(
+            "    ╰── {} python {}",
+            abi_mark, abi_output.python_abi_version
+        );
+    }
+}
+
+#[derive(Serialize)]
+pub struct RepoErrorResponse {
+    repository: String,
+    status: String,
+    error: String,
+}
+
+#[derive(Serialize)]
+pub struct RepositoryCheckResult {
+    repository: String,
+    status: String,
+    build_status: BuildStatus,
+    abi_status: AbiStatus,
+}
+
+#[derive(Serialize)]
+pub struct BuildStatus {
+    summary: String,
+    cuda: CudaStatus,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    rocm: Option<RocmStatus>,
+}
+
+#[derive(Serialize)]
+pub struct CudaStatus {
+    compatible: bool,
+    present: Vec<String>,
+    missing: Vec<String>,
+}
+
+#[derive(Serialize)]
+pub struct RocmStatus {
+    compatible: bool,
+    present: Vec<String>,
+    missing: Vec<String>,
+}
+
+#[derive(Serialize)]
+pub struct AbiStatus {
+    compatible: bool,
+    manylinux_version: String,
+    python_abi_version: String,
+    variants: Vec<VariantCheckOutput>,
+}
+
+#[derive(Serialize)]
+pub struct VariantCheckOutput {
+    name: String,
+    compatible: bool,
+    has_shared_objects: bool,
+    violations: Vec<String>,
+}
+
+pub fn get_cache_dir() -> Result<PathBuf> {
+    let cache_dir = if let Ok(dir) = std::env::var("HF_KERNELS_CACHE") {
+        PathBuf::from(dir)
+    } else {
+        dirs::home_dir()
+            .unwrap_or_else(std::env::temp_dir)
+            .join(".cache/huggingface/hub")
+    };
+
+    if !cache_dir.exists() {
+        fs::create_dir_all(&cache_dir).context("Failed to create cache directory")?;
+    }
+
+    Ok(cache_dir)
+}
+
+/// Get "org/name" repo ID from filesystem path
+pub fn get_repo_id_from_path(path: &Path) -> Result<String> {
+    // Extract the organization and model name from the path
+    let dir_name = path
+        .file_name()
+        .ok_or_else(|| CompliantError::Other(format!("Invalid path: {:?}", path)))?
+        .to_string_lossy()
+        .to_string();
+
+    // Remove the "models--" prefix if present
+    let dir_name = dir_name
+        .strip_prefix("models--")
+        .unwrap_or(&dir_name)
+        .replace("--", "/");
+
+    Ok(dir_name)
+}
+
+/// Check if repository has build variants
+pub fn has_build_variants(repo_path: &Path) -> Result<bool> {
+    // Look for the snapshot directory
+    let ref_file = repo_path.join("refs/main");
```

Comment:
Does not work for checking compliance of a kernel on a branch, which seems like a common thing to do: first check the kernel before merging.

Reply:
removed and now relying on hf-hub for file locations